_rules/button-accessible-name-97a4e1.md

```diff
@@ -172,26 +208,68 @@ This off screen `button` element has no [accessible name][] because it has no co
 </html>
 ```
 
+#### Failed Example 5
+
+This `input` element with 'type' attribute in the `Image button` state has an empty [accessible name][]. The `name` attribute can not be used to provide an [accessible name][].
+
+```html
+<input type="image" name="search" src="/test-assets/shared/search-icon.svg" />
+```
+
+#### Failed Example 6
+
+This image button has an empty `alt` attribute, and no other attributes that can give it an [accessible name][].
+
+```html
+<input type="image" src="/test-assets/shared/search-icon.svg" alt="" />
+```
+
+#### Failed Example 7
+
+This image button has an `aria-labelledby` attribute, but the referenced element does not exist. This gives the button an empty [accessible name][].
+
+```html
+<input type="image" src="/test-assets/shared/search-icon.svg" aria-labelledby="non-existing" />
+```
+
+
 ### Inapplicable
 
 #### Inapplicable Example 1
 
-This `input` element has a `type` attribute set to `image`. These images are tested in a separate rule which also tests [success criterion 1.1.1 Non-text Content](https://www.w3.org/TR/WCAG21/#non-text-content).
+This `button` element does not need an [accessible name][] because it is not included in the accessibility tree.
 
 ```html
-<input type="image" value="download" alt="Download" />
+<button style="display: none;"></button>
 ```
 
 #### Inapplicable Example 2
 
-This `button` element does not need an [accessible name][] because it is not included in the accessibility tree.
+
+This `input` element with 'type' attribute in the `Image button` state does not need an [accessible name][] because it is not included in the accessibility tree.
 
 ```html
-<button style="display: none;"></button>
+<input type="image" src="/test-assets/shared/search-icon.svg" style="display: none;" />
 ```
 
 #### Inapplicable Example 3
 
+The `button` element is tested separately from the `img` element. [Success Criterion 4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value) is applied to the button, whereas the image is tested under [Success Criterion 1.1.1 Non-text Content](https://www.w3.org/TR/WCAG21/#non-text-content)
+
+```html
+<button><img src="/test-assets/shared/search-icon.svg" alt="Search" /></button>
```

Comment:
The `button` element still has a semantic role of `button` and thus is applicable for the rule.

Reply:
Would this code ever be considered an image button? Or is it a failed example?

Reply:
This is not an image button as per the sense "`input` element in the `Image Button` state".
But this is the rule checking that elements with role of `button` have a (non empty) accessible name. In that case, there is an element with a role of `button` (the `button` element itself), thus the rule is applicable and passes because the `button` element gets its name from the content, here the, `alt` attribute of the image (iirc).